app/models/openscap_result.rb

```diff
@@ -0,0 +1,76 @@
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  def self.openscap_available?
+    # needed only for travis
+    require 'openscap'
+    require 'openscap/ds/arf'
+    require 'openscap/xccdf/benchmark'
+    true
+  rescue LoadError
+    false
+  end
+
+  def raw=(value)
+    raise "Cannot invoke raw more than once." unless binary_blob.nil?
```

Comment:
why can `raw=` be called only once?

I think you can express what you actually want better with validations
```ruby
validates :binary_blob, presence: true
```

Or do you really want to make sure, that once a blob and the resulting rule_results are saved they cannot be changed anymore?